sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -133,24 +134,24 @@ class AccountForgeMessageBuilder(
   }
 
   override def createNewBlock(
-                               nodeView: View,
-                               branchPointInfo: BranchPointInfo,
-                               isWithdrawalEpochLastBlock: Boolean,
-                               parentId: BlockId,
-                               timestamp: Timestamp,
-                               mainchainBlockReferencesData: Seq[MainchainBlockReferenceData],
-                               sidechainTransactions: Seq[SidechainTypes#SCAT],
-                               mainchainHeaders: Seq[MainchainHeader],
-                               ommers: Seq[Ommer[AccountBlockHeader]],
-                               ownerPrivateKey: PrivateKey25519,
-                               forgingStakeInfo: ForgingStakeInfo,
-                               vrfProof: VrfProof,
-                               forgingStakeInfoMerklePath: MerklePath,
-                               companion: DynamicTypedSerializer[SidechainTypes#SCAT, TransactionSerializer[SidechainTypes#SCAT]],
-                               inputBlockSize: Int,
-                               signatureOption: Option[Signature25519]): Try[SidechainBlockBase[SidechainTypes#SCAT, AccountBlockHeader]] = {
-
-
+      nodeView: View,
+      branchPointInfo: BranchPointInfo,
+      isWithdrawalEpochLastBlock: Boolean,
+      parentId: BlockId,
+      timestamp: Timestamp,
+      mainchainBlockReferencesData: Seq[MainchainBlockReferenceData],
+      sidechainTransactions: Seq[SidechainTypes#SCAT],
+      mainchainHeaders: Seq[MainchainHeader],
+      ommers: Seq[Ommer[AccountBlockHeader]],
+      ownerPrivateKey: PrivateKey25519,
+      forgingStakeInfo: ForgingStakeInfo,
+      vrfProof: VrfProof,
+      forgingStakeInfoMerklePath: MerklePath,
+      companion: DynamicTypedSerializer[SidechainTypes#SCAT, TransactionSerializer[SidechainTypes#SCAT]],
+      inputBlockSize: Int,
+      logsBloom: LogsBloom,
```

Comment:
we don't need to pass `logsBloom` as an argument here, because:
1) we don't know the value yet
2) it is applicable to the account model only.

Instead inside this method in this class we need to calculate a valid bloom filter using the real set of receipts (see `receiptList`)

Reply:
Good, so it needs to be calculated in 2 places? Here and in `AccountState` (it is already calculated there)-

Reply:
right